packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -374,6 +393,10 @@ abstract class ClusterBase extends Resource implements ICluster {
       targetType: secretsmanager.AttachmentTargetType.REDSHIFT_CLUSTER,
     };
   }
+
+  public addToParameterGroup(_name: string, _value: string): AddParameterResult {
+    return { parameterAddedResult: AddParameterResultStatus.IMPORTED_RESOURCE_FAILURE };
```

Comment:
For `addToParameterGroup` in the `Cluster` construct, this method should fail for all imported resources since we can't add parameters to them.  The L2 `Cluster` class is the only class that it wouldn't fail on. Any of the static import methods such as [fromClusterAttributes](https://github.com/dontirun/aws-cdk/blob/f2d50155a32cca0e5ad4b77002e2a9928794aaa9/packages/%40aws-cdk/aws-redshift/lib/cluster.ts#L389) should fail since they are imported resources. In the future there could *potentially*  be more static methods that create imported resources from `ClusterBase` which inherit this method but I don't think there would be another instantiated Cluster class that extends `ClusterBase`; so I thought the default should be the failure.

Reply:
Why can't we add parameters to imported Clusters?

Reply:
In CloudFormation, the Cluster is has a property to specify a [Parameter Group name](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-redshift-cluster.html#cfn-redshift-cluster-clusterparametergroupname) . [Parameter Groups](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-redshift-clusterparametergroup.html) don't have a property to tie them to a Cluster. That means that the association must be done on the side of the Cluster. I didn't think we could modify imports like that.

Additionally if we could modify imports, we could be potentially overriding an existing association since a Cluster can only have [one parameter group](https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-parameter-groups.html#about-parameter-groups) associated with it at a time